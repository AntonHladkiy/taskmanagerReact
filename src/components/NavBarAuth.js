import React from "react";
import {Link} from "react-router-dom";
const NavBarAuth = props => {
    return (
    <div>
        <Link to ="/signup"><button  className={"btn btn-info mt-2"}>Sign Up</button></Link>
        {(!props.loggedIn)&&<Link to ="/login"> <button  className="btn btn-info mt-2 ">Log In</button></Link>}
        {props.loggedIn&&<Link to ="/"> <button onClick={props.logOut} className="btn btn-danger mt-2 ">Log Out</button></Link>}
    </div>
    )
}
export default NavBarAuth;