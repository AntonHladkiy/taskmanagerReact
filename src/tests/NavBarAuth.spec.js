import React from 'react';
import { shallow } from 'enzyme';
import NavBarAuth from '../components/NavBarAuth';

describe('NavBarAuth logged in', () => {
    let wrapped = shallow(<NavBarAuth loggedIn={true}></NavBarAuth>);
    it('should render the NavBarAuth Component correctly', () => {
        expect(wrapped).toMatchSnapshot();
    });
});
describe('NavBarAuth not logged in', () => {
    let wrapped = shallow(<NavBarAuth loggedIn={false}></NavBarAuth>);
    it('should render the NavBarAuth Component correctly', () => {
        expect(wrapped).toMatchSnapshot();
    });

});