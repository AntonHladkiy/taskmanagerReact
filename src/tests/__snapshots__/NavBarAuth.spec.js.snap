// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBarAuth logged in should render the NavBarAuth Component correctly 1`] = `
<header
  className="d-flex flex-column p-3 px-md-4 mb-3 bg-white border-bottom shadow-sm"
>
  <nav
    className="navbar navbar-light bg-light justify-content-between"
  >
    <a
      className="navbar-brand"
    >
      Task Manager
    </a>
    <span
      className="navbar-text "
    >
      <span
        className="mr-2"
      >
        <Link
          to="/"
        >
           Main page
        </Link>
      </span>
      <span
        className="mr-2"
      >
        <Link
          to="/"
        >
           Log Out
        </Link>
      </span>
      <Link
        to="/signup"
      >
        Sign Up
      </Link>
    </span>
  </nav>
</header>
`;

exports[`NavBarAuth not logged in should render the NavBarAuth Component correctly 1`] = `
<header
  className="d-flex flex-column p-3 px-md-4 mb-3 bg-white border-bottom shadow-sm"
>
  <nav
    className="navbar navbar-light bg-light justify-content-between"
  >
    <a
      className="navbar-brand"
    >
      Task Manager
    </a>
    <span
      className="navbar-text "
    >
      <span
        className="mr-2"
      >
        <Link
          to="/"
        >
           Main page
        </Link>
      </span>
      <span
        className="mr-2"
      >
        <Link
          to="/login"
        >
           Log In
        </Link>
      </span>
      <Link
        to="/signup"
      >
        Sign Up
      </Link>
    </span>
  </nav>
</header>
`;
